File: nodes/array-reverse/array-reverse.process.py
async def process(inputs, settings, config, nodeConfig):
  array = inputs.get("array", [])
  if not isinstance(array, list):
    array = [array]
    
  reversed_array = array[::-1]
  indices = list(range(len(array) - 1, -1, -1))
  
  return {
    "array": reversed_array,
    "indices": indices
  } 

File: sdks/nodejs/nodes/array-reverse/array-reverse.process.js
export default async ({inputs, settings, config}) => {

  const array = Array.isArray(inputs.array) ? inputs.array : [inputs.array];
  
  const reversed = [...array].reverse();
  const indices = array.map((_, i) => array.length - 1 - i);
  
  return {
    array: reversed,
    indices: indices
  };
}; 